tests/st/calicoctl/test_default_pools.py

```diff
@@ -0,0 +1,168 @@
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", 2, "off", "IPv6, IPIP explicitly off"),
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/122", 0, "junk", "Invalid IPIP value"),
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", 0, "reboot", "Invalid IPIP value"),
+        (True, "CALICO_IPV4POOL_CIDR", "0.0.0.0/0", 2, None, "Valid, but not sensible"),
+        (True, "CALICO_IPV6POOL_CIDR", "::/0", 2, None, "Valid, but not sensible"),
+        (True, "CALICO_IPV6POOL_CIDR", "fd80::0:0/120", 2, None, "Valid, but non-canonical form"),
+        (False, "CALICO_IPV6POOL_CIDR", "1.2.3.4/24", 0, None, "Wrong type"),
+        (False, "CALICO_IPV4POOL_CIDR", "fd00::/24", 0, None, "Wrong type"),
+        (True, "CALICO_IPV6POOL_CIDR", "::0:a:b:c:d:e:0/120", 2, None, "Valid, non-canonical form"),
+        (False, "CALICO_IPV4POOL_CIDR", "1.2/16", 0, None, "Valid, unusual form"),
+    ])
+    def test_default_pools(self, success_expected, param, value, exp_num_pools, ipip, description):
+        """
+        Test that the various options for default pools work correctly
+        """
+        _log.debug("Test description: %s", description)
+        # Get command line for starting docker
+        output = self.host.calicoctl("node run --dryrun")
+        base_command = output.split('\n')[-4].rstrip()
+
+        # Modify command line to add the options we want to test
+        # Command line will attempt to use quay.io/calico/node:latest - we want calico/node:latest
+        base_command = base_command.replace("quay.io/", "")
+
+        env_inserts = "-e %s=%s " % (param, value)
+        if ipip is not None:
+            env_inserts += "-e CALICO_IPV4POOL_IPIP=%s " % ipip
+        prefix, _, suffix = base_command.partition("-e")
+        command = prefix + env_inserts + "-e" + suffix
+
+        # Start calico-docker
+        self.host.execute(command)
+
+        if not success_expected:
+            # check for "Calico node failed to start"
+            expected_string = "Calico node failed to start"
+            check = functools.partial(self.assert_calico_node_log_contains, expected_string)
+            retry_until_success(check, 5, ex_class=AssertionError)
+            return
+
+        # Check we started OK
+        expected_string = "Calico node started successfully"
+        check = functools.partial(self.assert_calico_node_log_contains, expected_string)
+        retry_until_success(check, 5, ex_class=AssertionError)
+        # check the expected pool is present
+        pools_output = self.host.calicoctl("get ippool -o yaml")
+        pools_dict = yaml.safe_load(pools_output)
+        cidrs = [pool['metadata']['cidr'] for pool in pools_dict]
+        # Convert to canonical form
+        value = str(netaddr.IPNetwork(value))
+        assert value in cidrs, "Didn't find %s in %s" % (value, cidrs)
+
+        # Dump pools and attempt to load them with calicoctl (to confirm consistency)
+        self.host.calicoctl("get ippool -o yaml > pool")
```

Comment:
I think this will write out to the Dockerhost's file system and get destroyed with the host.  I'm certainly not seeing this file in my local file system?